newIDE/app/src/AssetStore/CreditsPackages/CreditsPackageStoreContext.js

```diff
@@ -81,18 +118,112 @@ export const CreditsPackageStoreStateProvider = ({
+  );
+
+  const suggestedPackage: ?CreditsPackageListingData = React.useMemo(
+    () => {
+      if (!missingCredits || !creditsPackageListingDatas) return null;
+      let creditsPackageListingDataWithShorterPositiveDifferenceInCredits = null;
+      creditsPackageListingDatas.forEach(creditsPackageListingData => {
+        const packageCreditsAmount = getCreditsAmountFromId(
+          creditsPackageListingData.id
+        );
+        const selectedPackageCreditsAmount = creditsPackageListingDataWithShorterPositiveDifferenceInCredits
+          ? getCreditsAmountFromId(
+              creditsPackageListingDataWithShorterPositiveDifferenceInCredits.id
+            )
+          : null;
+        if (
+          packageCreditsAmount > missingCredits &&
+          (!selectedPackageCreditsAmount ||
+            packageCreditsAmount - missingCredits <
+              selectedPackageCreditsAmount - missingCredits)
+        ) {
+          creditsPackageListingDataWithShorterPositiveDifferenceInCredits = creditsPackageListingData;
+        }
+      });
+
+      // If no package with more credits than missingCredits is found, return the package with the most credits.
+      if (!creditsPackageListingDataWithShorterPositiveDifferenceInCredits) {
+        creditsPackageListingDataWithShorterPositiveDifferenceInCredits = creditsPackageListingDatas.reduce(
+          (a, b) => {
+            return getCreditsAmountFromId(a.id) > getCreditsAmountFromId(b.id)
+              ? a
+              : b;
+          }
+        );
+      }
+
+      return creditsPackageListingDataWithShorterPositiveDifferenceInCredits;
+    },
+    [creditsPackageListingDatas, missingCredits]
+  );
+
+  const closeCreditsPackageDialog = React.useCallback(() => {
+    setIsCreditsPackageDialogOpen(false);
+  }, []);
+
+  const openCreditsUsageDialog = React.useCallback(
+    (options: CreditsUsageDialogOptions) => {
+      setCreditsUsageDialogConfig(options);
+      setIsCreditsUsageDialogOpen(true);
+    },
+    []
+  );
+
   const CreditsPackageStoreState = React.useMemo(
     () => ({
       creditsPackageListingDatas,
       fetchCreditsPackages,
+      openCreditsPackageDialog,
+      closeCreditsPackageDialog,
       error,
+      openCreditsUsageDialog,
     }),
-    [creditsPackageListingDatas, error, fetchCreditsPackages]
+    [
+      creditsPackageListingDatas,
+      fetchCreditsPackages,
+      openCreditsPackageDialog,
+      closeCreditsPackageDialog,
+      error,
+      openCreditsUsageDialog,
+    ]
   );
 
   return (
     <CreditsPackageStoreContext.Provider value={CreditsPackageStoreState}>
       {children}
+      {isCreditsPackageDialogOpen && (
+        <CreditsPackagesDialog
+          onClose={() => setIsCreditsPackageDialogOpen(false)}
+          suggestedPackage={suggestedPackage}
+          missingCredits={missingCredits}
+          showCalloutTip={showCalloutTip}
+        />
+      )}
+      {isCreditsUsageDialogOpen && creditsUsageDialogConfig && (
```

Comment:
If `creditsUsageDialogConfig` is mandatory, you could get rid of the `isCreditsUsageDialogOpen` variable

Reply:
nice catch